ansible/lookup_plugins/lookup/aws_regions.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+
+try:
+    from botocore.session import get_session
+    HAS_BOTOCORE = True
+except ImportError:
+    HAS_BOTOCORE = False
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables=None, **kwargs):
+
+        if not HAS_BOTOCORE:
+            raise AnsibleError("The aws_regions lookup requires the botocore library")
+
+        loader = get_session().get_component('data_loader')
```

Comment:
This is functional, but I'm going to update this to use boto3.Session().get_available_regions() instead. There are some "undocumented" regions that this current code picks up, but it appears the endpoints are either unaccessible or actually use the same endpoint as documented regions, so using get_available_regions() is simpler.

Reply:
this just came up for my other usages of this api, theres a new region enablement mechanism thats done in the account starting with the hong kong region, which makes new regions opt-in. There's an ec2.describe_regions api call that should be used preferentially as the sdk pickups all regions.